presto-main/src/test/java/io/prestosql/execution/TestSqlTaskExecution.java

```diff
@@ -209,6 +212,7 @@ public void testSimple(PipelineExecutionStrategy executionStrategy)
                     // assert that task result is produced
                     outputBufferConsumer.consume(300 + 200, ASSERT_WAIT_TIMEOUT);
                     outputBufferConsumer.assertBufferComplete(ASSERT_WAIT_TIMEOUT);
+                    assertEquals(taskStateMachine.getStateChange(RUNNING).get(10, SECONDS), FLUSHING);
```

Comment:
shouldn't that be moved above `outputBufferConsumer.consume(300 + 200, ASSERT_WAIT_TIMEOUT);` to prevent race condition?

Does `outputBufferConsumer.assertBufferComplete(ASSERT_WAIT_TIMEOUT);` consume all output?

Reply:
Yes, `assertBufferComplete` makes sure that the buffer is complete. Added the assertions as discussed offline.